lib/swaps/Swaps.ts

```diff
@@ -951,13 +951,30 @@ class Swaps extends EventEmitter {
         if (err.code === errorCodes.UNKNOWN_PAYMENT_ERROR) {
           // the payment failed but we are unsure of its final status, so we fail
           // the deal and assign the payment to be checked in swap recovery
-          clearTimeout(this.timeouts.get(deal.rHash));
-          this.timeouts.delete(deal.rHash);
-          this.emit('swap.failed', deal);
-          this.deals.delete(deal.rHash);
-          const swapDealInstance = await this.repository.getSwapDeal(rHash);
-          this.swapRecovery.pendingSwaps.add(swapDealInstance!);
+          // we cannot remove our incoming invoice because we are not yet certain
+          // whether our outgoing payment can be claimed by the taker or not
+          await this.failDeal({
+            deal,
+            peer,
+            failedCurrency: deal.takerCurrency,
+            failureReason: SwapFailureReason.UnknownError,
+            errorMessage: err.message,
+          });
+
+          // we may already be in swap recovery for this deal due to a timeout
+          // prior to the payment failing for an unknown reason. if not, we
+          // put this deal into swap recovery right now to monitor for a
+          // conclusive resolution
+          if (!this.swapRecovery.pendingSwaps.has(rHash)) {
+            const swapDealInstance = await this.repository.getSwapDeal(rHash);
+            this.swapRecovery.pendingSwaps.set(rHash, swapDealInstance!);
+          }
         } else {
+          // the payment failed permanently, so we remove our incoming invoice
+          // and fail the deal
+          const makerSwapClient = this.swapClientManager.get(deal.makerCurrency)!;
+          makerSwapClient.removeInvoice(deal.rHash).catch(this.logger.error); // we don't need to await the remove invoice call
```

Comment:
Looks like the current `sendPayment` error is `7.4 - Timeout has occurred` so it's getting to this block.